tools/dev_cluster.py

```diff
@@ -127,6 +172,26 @@ async def run(self):
         await self.process.wait()
 
 
+async def ensure_bucket_exists(cfg: RedpandaConfig):
+    session = aioboto3.Session()
+    client = session.client(
+        service_name="s3",
+        endpoint_url=
+        f"http://{cfg.cloud_storage_api_endpoint}:{cfg.cloud_storage_api_endpoint_port}",
+        aws_access_key_id=cfg.cloud_storage_access_key,
+        aws_secret_access_key=cfg.cloud_storage_secret_key)
+    print("Preparing cloud storage")
+    async with client as s3:
+        while True:
+            buckets = await s3.list_buckets()
```

Comment:
Can this happen before minio is ready and throw? or are there retires behind the scenes?

Reply:
yeh, fixed